src/rust/ensogl/lib/core/src/display/shape/primitive/system.rs

```diff
@@ -141,6 +141,10 @@ pub trait Shape : display::Object + CloneRef + Debug + Sized {
     type System : ShapeSystemInstance<Shape=Self>;
     /// Accessor for the underlying sprites.
     fn sprites(&self) -> Vec<&Sprite>;
+    /// Set the canvas size of this shape.
+    fn set_size(&self, size:Vector2<f32>) {
+        self.sprites().iter().for_each(move |sprite| sprite.size.set(size))
+    }
```

Comment:
Please revert this change, as it will clash with my branch and I'll have hard time merging it. On my branch I'm redefining a lot of shape system handling, so it would be much easier for me if we dont intorduce api changes until its merged